### PR TITLE
changes to support pubmatic bidder in permutive module

### PR DESCRIPTION
**Pull request by pm-manasi-moghe**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

File: modules/permutiveRtdProvider.js [hunk 1]
```diff
@@ -125,6 +127,43 @@ function getDefaultBidderFn (bidder) {
...
+        segments = existingData ? segments.concat(existingData) : [];
+        // add AC specific segments
+        if (acEnabled && data.ac && data.ac.length > 0) {
+          for (var index in data.ac) {
+            segments = segments.concat({id: data.ac[index]});
+          }
+        }
+
+        if (!dataUpdated && segments && segments.length > 0) {
+          window['pbjs'].setBidderConfig({
```
Comment (pm-shashank-jain) [inline modules/permutiveRtdProvider.js:151]: namespace can be changed by publisher so we can use global parameter for namespace

Comment (pm-shashank-jain) [reply modules/permutiveRtdProvider.js:151]: Do we need to set the bidderConfig, can't we add it in Bid Object just like other partners?

File: modules/permutiveRtdProvider.js [hunk 2]
```diff
@@ -125,6 +127,43 @@ function getDefaultBidderFn (bidder) {
       }
 
       return bid
+    },
+    pubmatic: function(bid, data, acEnabled) {
+      let segments = [];
+      var existingData = config.getBidderConfig(); // read existing bidder level configs if any
```
Comment (pm-shashank-jain) [inline modules/permutiveRtdProvider.js:133]: Can we handle merging of data in PubMaticAdapter and here can we just read data from permutative and send it to bidder.

File: modules/permutiveRtdProvider.js [hunk 3]
```diff
@@ -186,4 +225,8 @@ export const permutiveSubmodule = {
   init: init
 }
 
+function getNestedData(obj, ...args) {
```
Comment (pm-shashank-jain) [inline modules/permutiveRtdProvider.js:228]: can this function be moved to utils ?

**Comment by pm-shashank-jain** [comment]
We should add unit test cases before raising PR to prebid.

File: modules/permutiveRtdProvider.js (hunk 1 above)
Comment (pm-manasi-moghe) [reply modules/permutiveRtdProvider.js]: namespace change done.

File: modules/permutiveRtdProvider.js (hunk 2 above)
Comment (pm-manasi-moghe) [reply modules/permutiveRtdProvider.js]: its kept here, since its permutive specific.

File: modules/permutiveRtdProvider.js (hunk 3 above)
Comment (pm-manasi-moghe) [reply modules/permutiveRtdProvider.js]: done.

**Pull request closed without merging**